common/models/user.js

```diff
@@ -448,21 +448,28 @@ module.exports = function(User) {
     var Email =
       options.mailer || this.constructor.email || registry.getModelByType(loopback.Email);
 
-    // Set a default token generation function if one is not provided
-    var tokenGenerator = options.generateVerificationToken || User.generateVerificationToken;
-
-    tokenGenerator(user, function(err, token) {
-      if (err) { return fn(err); }
-
-      user.verificationToken = token;
+     // Set a default token generation function if one is not provided    
+    if(!options.generateVerificationToken){
+      User.generateVerificationToken(user, function(err, token) {
+        if (err) { return fn(err); }
+        user.verificationToken = token;
+        saveAndSend(user);
+      })   
+    } else {
+      user.verificationToken = options.generateVerificationToken;
```

Comment:
Hmm, do you expect `options.generateVerificationToken` to be the value for `verificationToken`? If so, that is wrong. `options.generateVerificationToken` should be a JS function that is capable of generating verification tokens for a user.